lib/source/metadata.js

```diff
@@ -87,24 +87,59 @@ class Metadata extends Source.Polling(Parser) { // eslint-disable-line new-cap
           return callback(err);
         }
 
-        // Detect change by hashing the fetched data
-        const hash = Crypto.createHash('sha1');
-        const paths = Object.keys(data);
+        let p = Promise.resolve(data);
+
+        // Grab the ASG from the instance-id
+        const instanceId = data['meta-data/instance-id'];
+        const az = data['meta-data/placement/availability-zone'];
+
+        if (instanceId && az) {
+          const region = az.slice(0, -1);
+
+          p = new Promise((resolve, reject) => {
+            (new Aws.AutoScaling({region})).describeAutoScalingInstances({InstanceIds: [instanceId]}, (err, d) => {
+              if (err) {
+                return reject(err);
+              }
+              resolve(d);
+            });
+          }).then((d) => {
+            const asg = d.AutoScalingInstances.map((instance) => instance.AutoScalingGroupName);
+
+            // No reason it should be longer than 1 but worth a check
+            if (asg.length > 1) {
+              Log.log('WARN', `Instance id ${instanceId} is in multiple auto-scaling groups`, asg);
+            }
+
+            // Check to see if an instance is actually part of an ASG
+            if (asg.length !== 0) {
+              data['auto-scaling-group'] = asg[0];
+            }
+
+            return data;
+          }).catch(() => data);
```

Comment:
Is this `catch` going to swallow the error that a failed `describeAutoScalingInstances` call might throw?